src/cpu/x64/matmul/amx_blocking_heuristics.cpp

```diff
@@ -570,6 +570,37 @@ bool matmul_amx_blocking_params_macro_t::is_horizontal_selected(
     return is_horizontal_local;
 }
 
+void mem_advice_init(brgemm_matmul_conf_t &bgmmc) {
```

Comment:
1. Class specification?
```suggestion
void matmul_amx_blocking_params_macro_t::mem_advice_init(brgemm_matmul_conf_t &bgmmc) {
```

2. Why this is a part of blocking heuristics and not a part of `bgmmc` as the code below clearly updates it with some arguments and doesn't seem to rely on anything but M_chunks and N_chunks... None of other methods allow to modify descriptor in this class.

Reply:
Removing mem_advice_init to `matmul_utils`.